torch/__init__.py

```diff
@@ -141,25 +141,38 @@ def _preload_cuda_deps():
     """Preloads cudnn/cublas deps if they could not be found otherwise."""
     # Should only be called on Linux if default path resolution have failed
     assert platform.system() == 'Linux', 'Should only be called on Linux'
-    cublas_path = None
-    cudnn_path = None
+
+    cuda_libs: Dict[str, str] = {
+        'cublas': 'libcublas.so.11',
+        'cudnn': 'libcudnn.so.8',
+        'cuda_nvrtc': 'libnvrtc.so.11.2',
```

Comment:
Sorry, why do we need to preload nvrtc here? (Aren't it supposed to be loaded lazily anyway)

Reply:
@syed-ahmed @malfet I think lazy loading was enabled by this PR if I am not mistaken: https://github.com/pytorch/pytorch/pull/85692 . However the question, will lazy loading will find this dll if we don't preload ?

Reply:
libnvrtc.so.11.2 is listed as a dependency for libcaffe2_nvrtc:
```
ldd libcaffe2_nvrtc.so
        linux-vdso.so.1 (0x00007ffe179d6000)
        libcuda.so.1 => /usr/lib/x86_64-linux-gnu/libcuda.so.1 (0x00007fcb8a61d000)
        libnvrtc.so.11.2 => not found
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fcb8a294000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fcb89ef6000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fcb89cde000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fcb898ed000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fcb896e9000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fcb894ca000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fcb892c2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fcb8ba81000)
```
Also @atalman's concern is valid I think. Where is nvrtc lazily loaded in pytorch? (I don't think [CUDA_MODULE_LOADING](https://docs.nvidia.com/cuda/cuda-c-programming-guide/lazy-loading.html?highlight=cuda_module_loading#what-is-lazy-loading) has anything to do with lazy loading of dlls, and it's for lazy loading of kernels)